improver_tests/regrid/test_RegridWithLandSeaMask.py

```diff
@@ -334,3 +336,59 @@ def test_regrid_bilinear_with_mask_2():
     np.testing.assert_allclose(
         regrid_bilinear_with_mask.data, expected_results, atol=1e-3
     )
+
+
+@pytest.mark.parametrize("regridder", ("nearest", "bilinear"))
+@pytest.mark.parametrize("landmask", (True, False))
+@pytest.mark.parametrize("maskedinput", (True, False))
+def test_target_domain_bigger_than_source_domain(regridder, landmask, maskedinput):
+    """Test regridding when target domain is bigger than source domain"""
+
+    # set up source cube, target cube and land-sea mask cube
+    cube_in, cube_out_mask, cube_in_mask = define_source_target_grid_data_same_domain()
+
+    # add a circle of grid points so that output domain is much bigger than input domain
+    width_x, width_y = 2, 4  # lon,lat
+    cube_out_mask_pad = pad_cube_with_halo(cube_out_mask, width_x, width_y)
+
+    if landmask:
+        with_mask = "-with-mask"
+    else:
+        with_mask = ""
+        cube_in_mask = None
+    regrid_mode = f"{regridder}{with_mask}-2"
+
+    if maskedinput:
+        # convert the input data to a masked array with no values covered by the mask
+        cube_in_masked_data = np.ma.masked_array(cube_in.data, mask=False)
+        cube_in.data = cube_in_masked_data
+
+    # run the regridding
+    regridder = RegridLandSea(
+        regrid_mode=regrid_mode, landmask=cube_in_mask, landmask_vicinity=250000000,
+    )
+    regrid_out = regridder(cube_in, cube_out_mask)
+    regrid_out_pad = regridder(cube_in, cube_out_mask_pad)
+
```

Comment:
Please add a test to check that the output data are of the same dtype as the input data (both should be `float32`)

Reply:
@MoseleyS I have come back from leave. Thank you very much for your detailed comments. 
Some groups in BOM are interested in using the land-sea-mask-aware regridding, and I would like to recommend them to install IMPROVER and then use regridding function. I thought that to make the input and output array type consistent would be good. 
I will do what you suggested. Thanks,
Julian